.github/workflows/backend.yml

```
@@ -23,7 +23,7 @@ jobs:
     - name: Install Go
       uses: actions/setup-go@v4
       with:
-        go-version: 1.20.x
+        go-version: 1.21.x
```

It might make sense to use something like this instead:
```suggestion
        go-version-file: './go.mod'
```
This would reduce the number of places where we have to upgrade Go in the future. 
Obviously there are multiple `go.mod`s in this repo, so if there is one that makes more sense than the one in the root of the repository, it could be changed to that instead

---

Hey @AndersBennedsgaard 
I think these are very valid suggestions, but they are out of scope of this PR which is merely to upgrade the go version. My suggestion is to create an issue for this or propose a separate PR.